lua/mason-lspconfig/server_configurations/drools_lsp/init.lua

```diff
@@ -0,0 +1,21 @@
+return function()
+    return {
+        on_new_config = function(new_config)
+            -- if the user configured a cmd, we just use that
+            if not new_config.cmd then
+                -- otherwise, we look to see if they set the location of the drools jar
+                local drools_jar = vim.tbl_get(new_config.settings, "drools", "jar")
+                        and new_config.settings["drools"]["jar"]
+                    or nil
+                if drools_jar ~= nil then
+                    -- if the drools jar location was set, we re-use the logic in the default_config
+                    local default_config = require("lspconfig.server_configurations.drools_lsp").default_config
+                    return default_config.on_new_config(new_config)
+                else
+                    -- otherwise, we fall back to the shell_exec_wrapper that mason generated
+                    new_config.cmd = { "drools-lsp" }
+                end
+            end
+        end,
```

Comment:
I believe this logic could all be simplified into

```suggestion
        cmd = { "drools-lsp" }
```

Mason will merge these configurations in the order of 

```
user provided config > mason-lspconfig (this) > lspconfig default config
```

So any `cmd` the user provides would have precedence.

Reply:
Other than that let me know if this PR is ready to be merged or if you wanted to change something first!

Reply:
@williamboman Thanks for the detailed review. I don't have time right now to digest it properly and potentially make any changes, but I should be able to later today. So please leave this PR open and don't merge it yet. Thank you again!